### PR TITLE
Support for platforms based on Barefoot Networks' device

### PR DESCRIPTION
**Pull request by rsunkad**
**- Incorporate support for platforms based on Barefoot Networks' device **

**- Added port config files and related scripts for the platforms **

**- Verified using SONiC tests (requires the corresponding changes to sairedis and swss) **

p4

### CODE REVIEW

File: platform/barefoot/one-image.mk [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# sonic one image installer
+
+SONIC_ONE_IMAGE = sonic-barefoot.bin
+$(SONIC_ONE_IMAGE)_MACHINE = barefoot
+$(SONIC_ONE_IMAGE)_IMAGE_TYPE = onie
+$(SONIC_ONE_IMAGE)_INSTALLS += $(BFN_PLATFORM_MODULE) 
+$(SONIC_ONE_IMAGE)_INSTALLS += $(BFN_MONTARA_PLATFORM_MODULE)
+$(SONIC_ONE_IMAGE)_INSTALLS += $(WNC_OSW1800_PLATFORM_MODULE) 
+$(SONIC_ONE_IMAGE)_INSTALLS += $(INGRASYS_S9180_32X_PLATFORM_MODULE)
```
Comment (wadelnn) [inline platform/barefoot/one-image.mk]: The last three modules must use LAZY_INSTALL like:
`$(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(BFN_MONTARA_PLATFORM_MODULE)`
`$(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(WNC_OSW1800_PLATFORM_MODULE) `
`$(SONIC_ONE_IMAGE)_LAZY_INSTALLS += $(INGRASYS_S9180_32X_PLATFORM_MODULE)`
Or you will install all platform drivers in each platform and platform driver will init failed.
You can reference PR as follows https://github.com/barefootnetworks/sonic-buildimage/pull/6

File: platform/barefoot/sonic-platform-modules-wnc-osw1800/debian/files [hunk 2]
```diff
@@ -0,0 +1 @@
+platform-modules-wnc-osw1800_1.0_amd64.deb main extra
```
Comment (lguohan) [inline platform/barefoot/sonic-platform-modules-wnc-osw1800/debian/files]: this is debian generated file, we need to remove.

File: .gitmodules [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 	path = src/sonic-swss-common
-	url = https://github.com/Azure/sonic-swss-common
+	url = https://github.com/barefootnetworks/sonic-swss-common
 [submodule "sonic-linux-kernel"]
 	path = src/sonic-linux-kernel
 	url = https://github.com/Azure/sonic-linux-kernel
 [submodule "sonic-sairedis"]
 	path = src/sonic-sairedis
-	url = https://github.com/Azure/sonic-sairedis
+	url = https://github.com/barefootnetworks/sonic-sairedis
```
Comment (lguohan) [inline .gitmodules]: can you open PR against sonic-sairedis repo on 201712 branch, we'll take the patch at that repo, then update the reference here.

File: .gitmodules [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
 [submodule "sonic-swss"]
 	path = src/sonic-swss
-	url = https://github.com/Azure/sonic-swss
+	url = https://github.com/barefootnetworks/sonic-swss
```
Comment (lguohan) [inline .gitmodules]: can you open PR on sonic-swss 201712 branch, right now, i am not sure I can take all patches you have made, but we can discuss there one-by-one.

File: platform/barefoot/sonic-platform-modules-bfn-montara/debian/files [hunk 5]
```diff
@@ -0,0 +1 @@
+platform-modules-bfn-montara_1.0_amd64.deb main extra
```
Comment (lguohan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/debian/files]: auto-generated file

File: .gitmodules [hunk 6]
```diff
@@ -1,15 +1,21 @@
 [submodule "sonic-swss-common"]
 	path = src/sonic-swss-common
-	url = https://github.com/Azure/sonic-swss-common
+	url = https://github.com/barefootnetworks/sonic-swss-common
```
Comment (lguohan) [inline .gitmodules]: can you change this back to Azure submodule

**Comment by lguohan** [comment]
I have merge the commits in swss and sairedis repo, can you change your submodule to Azure repo? 201712 branch?

**Comment by lguohan** [comment]
looks like sonic-swss-common ref point is still updated in this PR, it seems not needed, can you check.

File: dockers/docker-bfn-sswsyncd-rpc/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,38 @@
+FROM docker-base
+
+## Make apt-get non-interactive
+ENV DEBIAN_FRONTEND=noninteractive
+
```
Comment (lguohan) [inline dockers/docker-bfn-sswsyncd-rpc/Dockerfile]: per our discussion, this docker should be removed.

File: dockers/docker-bfn-sswsyncd/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,17 @@
+FROM docker-base
+
```
Comment (lguohan) [inline dockers/docker-bfn-sswsyncd/Dockerfile]: remove this docker.

File: dockers/docker-orchagent/swssconfig.sh [hunk 9]
```diff
@@ -42,6 +42,10 @@ elif [[ "$HWSKU" == "ACS-MSN27"* ]]; then
     SWSSCONFIG_ARGS+="msn27xx.32ports.buffers.json "
 fi
 
+if [ "$HWSKU" != "montara" ] && [ "$HWSKU" != "mavericks" ] && [ "$HWSKU" != "OSW1800-48x6q" ] && [ "$HWSKU" != "INGRASYS-S9180-32X"]; then
+    SWSSCONFIG_ARGS+="ipinip.json "
+fi
```
Comment (lguohan) [inline dockers/docker-orchagent/swssconfig.sh]: can you keep the order? keep ipinip.json between copp and ports.

**Pull request merged**